src/RingGrids/grids/octahedral_clenshaw.jl

```diff
@@ -107,12 +135,47 @@ function Matrix!(   MGs::Tuple{AbstractMatrix{T}, OctahedralClenshawGrid}...;
             r += (sr-1)*nside                       # shift row into matrix quadrant
             c += (sc-1)*nside                       # shift column into matrix quadrant
 
-            for (Mi, Gi) in MGs                      # for every (matrix, grid) tuple
-                Mi[r, c] = convert(T, Gi[ij])         # convert data and copy over
+            for (Mi, Gi) in MGs                     # for every (matrix, grid) tuple
+                Mi[r, c] = convert(T, Gi[ij])       # convert data and copy over
```

Comment:
This wouldn't work on GPU, and we'd have to think about another version. We'd have to check also other conversion routine. However, initially it's probably fine to define only those that matter for the dynamical core. If it is e.g. only used for the output, then it's fine to have it CPU only, as the output always has to be on CPU anyway.

Reply:
Yes the `Matrix!` conversion is some optional output that I used at some point to create these area-preserving projections

![matrix_output](https://github.com/SpeedyWeather/SpeedyWeather.jl/assets/25530332/dd49ba64-eedb-4421-b813-98ca511805d3)

So I'd just not touch it for now, it's not high priority for GPU stuff.

Reply:
Okay, agreed!